src/common/errors.ts

```diff
@@ -0,0 +1,3 @@
+export interface ErrorWithStatus extends Error {
```

Comment:
ErrorWithStasus sounds wired, I would change it to HttpError

Reply:
I agree with Gal

Reply:
i was overruled.
done